core/src/main/java/org/apache/iceberg/deletes/EqualityDeleteWriter.java

```diff
@@ -58,8 +60,9 @@ public EqualityDeleteWriter(FileAppender<T> appender, FileFormat format, String
   }
 
   @Override
-  public void write(T row) {
+  public PathOffset write(T row) {
     appender.add(row);
+    return PathOffset.of(location, recordCount++);
```

Comment:
In Iceberg, we don't use the return value of `++` operators because it is hard to read code that uses them. Can you move the increment to a separate line?

Reply:
Changed to:
```java
    long offset = recordCount++;
    return PathOffset.of(location, offset);
```